src/to_do/deque_list/__init__.py

```diff
@@ -0,0 +1,57 @@
+import typing
+from dataclasses import dataclass, field
+
+from src.to_do.base_list import BaseListTask
+from collections import deque
+from src.to_do.deque_list.task import DequeTask, task_01, task_02, task_03
+
+
+@dataclass
+class DequeList(BaseListTask):
+    tasks: typing.Deque[DequeTask] = field(default_factory=deque)
+
+    def add_task(self, task: DequeTask) -> None:
+        self.tasks.append(task)
+        task.status = 'PENDING'
+
+    def get_task(self) -> DequeTask:
+        if len(self.tasks) == 0:
+            return None
+        self.tasks[0].status = 'TAKEN'
+        return self.tasks.popleft()
+
+    def remove_task(self, task: DequeTask) -> None:
+        if len(self.tasks) == 0:
+            return None
+        self.tasks.remove(task)
+
+    def mark_task_as_completed(self, task: DequeTask) -> typing.List[DequeTask]:
+        if len(self.tasks) == 0:
+            return None
+        done_list = []
```

Comment:
`done_list` powinno być zainicjalizowane na poziomie klasy. Dokładnie obok `tasks`, na przykład w ten sposób:
```python
done_list: typing.List[DequeTask] = field(default_factory=list)
```
W Twoim kodzie, `done_list` będzie tylko lokalnie i przy kolejnym za każdym razem będzie na nowo pusta, a chcielibyśmy mieć pełną informację o zakończonych zadaniach.

Reply:
Dopowiadając, flow tej listy powinno wyglądać tak:
```python
# inicjalizacja listy
user_list1 = DequeList('Obowiazki')

# dodawanie tasków
user_list1.add_task(task_01)
user_list1.add_task(task_02)
user_list1.add_task(task_03)

# pobieranie taska do zrobienia
actual_task = user_list1.get_task()

# zakończenie pracy nad taskiem
user_list1.mark_task_as_completed(actual_task)
```

Trochę abstrakcyjne, ale tak to by działało w rzeczywistym świecie. Nie musimy też wtedy sprawdzać, czy task jest na liście czy nie, po prostu wrzucamy go na listę wykonanych 😉